### PR TITLE
Pack .cs files in .Sources packages as contentFiles

### PR DESCRIPTION
**Pull request by natemcmaster**
Puts *.cs files in contentFiles/cs/netstandard1.0 as buildAction="compile"
Puts *.resx files in contentFiles/any/netstandard1.0 as buildAction="EmbeddedResource"

~~Also puts *.cs files in shared/ for backwards compatibility with project.json.~~

cc @NTaylorMullen @anurse

### CODE REVIEW

File: makefile.shade [hunk 1]
```diff
@@ -1,91 +1,25 @@
...
+        foreach (var dir in Directory.GetDirectories(Directory.GetCurrentDirectory(), "src/*.Sources"))
+        {
+            var nugetPath = E("KOREBUILD_NUGET_EXE") ?? ".build/NuGet.exe";
+            var outputDir = "artifacts/build/";
+            var packageVersion = "1.2.0-" + E("DOTNET_BUILD_VERSION");
+            NugetPack(nugetPath,
+                "sources.nuspec",
+                outputDir,
+                packageVersion,
+                "-BasePath " + dir + " -Properties id=" + Path.GetFileName(dir));
```
Comment (Eilon) [inline makefile.shade]: Need to escape `dir` here in case it has spaces? Can it go in quotes or something?

File: test/Microsoft.Extensions.Internal.Test/project.json [hunk 2]
```diff
@@ -3,49 +3,25 @@
     "embed": {
       "include": "../../src/*.Sources/**/*.resx"
     },
-    "warningsAsErrors": true
+    "warningsAsErrors": true,
+    "compile": [
+      "../../src/Microsoft.Extensions.ClosedGenericMatcher.Sources/**/*.cs",
```
Comment (Eilon) [inline test/Microsoft.Extensions.Internal.Test/project.json]: Ehhhh........ can this continue to use packages? I'm worried that by not using the packages we won't detect problems in the packages themselves and the build will fail waaay later.

File: makefile.shade [hunk 3]
```diff
@@ -1,91 +1,25 @@
...
-      {
-          // Don't care if we couldn't delete the temp directory.
-      }
-   }
\ No newline at end of file
+#pack-sources target='compile'
+    @{
+        foreach (var dir in Directory.GetDirectories(Directory.GetCurrentDirectory(), "src/*.Sources"))
+        {
+            var nugetPath = E("KOREBUILD_NUGET_EXE") ?? ".build/NuGet.exe";
```
Comment (NTaylorMullen) [inline makefile.shade]: I believe there's a korebuild variable for the path to NuGet.exe

File: makefile.shade [hunk 4]
```diff
@@ -1,91 +1,25 @@
...
+            var outputDir = "artifacts/build/";
+            var packageVersion = "1.2.0-" + E("DOTNET_BUILD_VERSION");
```
Comment (NTaylorMullen) [inline makefile.shade]: Having this version just sit here in code may be difficult to 'remember' to update this on every release.

File: Common.sln [hunk 5]
```diff
@@ -8,22 +8,6 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "test", "test", "{6878D8F1-6
 EndProject
 Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "Microsoft.Extensions.Internal.Test", "test\Microsoft.Extensions.Internal.Test\Microsoft.Extensions.Internal.Test.xproj", "{D81EAC23-B515-49FB-B04B-285A9C1C6583}"
 EndProject
-Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "Microsoft.Extensions.CopyOnWriteDictionary.Sources", "src\Microsoft.Extensions.CopyOnWriteDictionary.Sources\Microsoft.Extensions.CopyOnWriteDictionary.Sources.xproj", "{D4BBCB55-914F-4BB3-922F-759FCC8C7728}"
```
Comment (NTaylorMullen) [inline Common.sln]: Why not at least migrate the projects to csproj? With how it is now we wont be able edit the source files in VS.

File: src/Microsoft.Extensions.CommandLineUtils/project.json [hunk 6]
```diff
@@ -10,7 +10,8 @@
     "nowarn": [
       "CS1591"
     ],
-    "xmlDoc": true
+    "xmlDoc": true,
+    "compile": "../Microsoft.Extensions.CommandLineUtils.Sources/**/*.cs"
```
Comment (NTaylorMullen) [inline src/Microsoft.Extensions.CommandLineUtils/project.json]: Curious: Why do we have two separate command line utils one as sources other as a package but both have the same content?

File: test/Microsoft.Extensions.Internal.Test/project.json [hunk 7]
```diff
@@ -3,49 +3,25 @@
...
-      "version": "1.2.0-*"
-    },
-    "Microsoft.Extensions.WebEncoders.Sources": {
-      "type": "build",
-      "version": "1.2.0-*"
-    },
     "Moq": "4.6.36-*",
-    "xunit": "2.2.0-*"
+    "xunit": "2.2.0-*",
+    "Microsoft.Extensions.PlatformAbstractions": "1.2.0-*"
```
Comment (NTaylorMullen) [inline test/Microsoft.Extensions.Internal.Test/project.json]: Why was this needed? Also same question for the security.claims below.

File: sources.nuspec [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
+  <metadata>
+    <id>$id$</id>
+    <version>$version$</version>
+    <authors>Microsoft</authors>
+    <owners>Microsoft</owners>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>$id$</description>
+    <contentFiles>
+      <files include="contentFiles\cs\any\**\*.cs" buildAction="Compile" />
```
Comment (pranavkm) [inline sources.nuspec]: Does this build on xplat?

**Comment by natemcmaster** [comment]
As both @Eilon and @NTaylorMullen pointed out, this PR intentionally removes the "projects" for the .Sources folders. This was a deliberate choice resulting from looking forward to MSBuild. AFAIK .NET Core SDK for MSBuild has no equivalent to the "shared" sources feature of project.json. See https://github.com/dotnet/sdk/issues/236.

File: test/Microsoft.Extensions.Internal.Test/project.json (hunk 7 above)
Comment (natemcmaster) [inline test/Microsoft.Extensions.Internal.Test/project.json]: Microsoft.Extensions.DotnetToolDispatcher.Sources requires Microsoft.Extensions.PlatformAbstractions.
Microsoft.Extensions.SecurityHelper.Sources requires System.Security.Claims.

File: src/Microsoft.Extensions.CommandLineUtils/project.json (hunk 6 above)
Comment (natemcmaster) [inline src/Microsoft.Extensions.CommandLineUtils/project.json]: The was created to enable a console tool "ef.exe" that has no dependencies. The same goal could be accomplished with ILMerge, but we decided a .Sources package was easier.

File: makefile.shade (hunk 4 above)
Comment (natemcmaster) [inline makefile.shade]: Will update.

File: test/Microsoft.Extensions.Internal.Test/project.json (hunk 7 above)
Comment (natemcmaster) [inline test/Microsoft.Extensions.Internal.Test/project.json]: The project.json for these apps includes these dependencies. We could preserve this behavior by putting it in the nuspec file, but we'd have to hard-code a dependency version since nuspec doesn't support floating dependency versions (-*). Instead, we'll need to update anyone using the .Sources package to include these additional dependencies.

**Comment by Eilon** [comment]
Talked to @natemcmaster and yeah this is fine... let's try to get a proper feature into MSBuild/NuGet for the future, but for now we'll make do with what's available.

**Comment by natemcmaster** [comment]
> let's try to get a proper feature into MSBuild/NuGet for the future

See https://github.com/NuGet/Home/issues/3683

**Comment by natemcmaster** [comment]
Ping. We okay to take this?

**Comment by NTaylorMullen** [comment]
:shipit:
